components/discussions/GithubDiscussions.tsx

```diff
@@ -0,0 +1,51 @@
+import GithubDiscussion from "@/components/discussions/GithubDiscussion";
+import { parseDateRangeSearchParam } from "@/lib/utils";
+import { ParsedDiscussion } from "@/scraper/src/github-scraper/types";
+
+interface Props {
+  discussions: ParsedDiscussion[];
+  searchParams?: { [key: string]: string };
+}
+
+const GithubDiscussions = ({ discussions, searchParams }: Props) => {
+  const category = searchParams?.category;
+  const dateRange = searchParams?.between;
+  const [start, end] = parseDateRangeSearchParam(dateRange);
+
+  if (category && dateRange) {
+    discussions = discussions.filter(
+      (discussion) =>
+        new Date(discussion.time) >= new Date(start) &&
+        new Date(discussion.time) <= new Date(end) &&
+        discussion.category?.name === category,
+    );
+  } else if (category) {
+    discussions = discussions.filter(
+      (discussion) => discussion.category?.name === category,
+    );
+  } else if (dateRange) {
+    discussions = discussions.filter(
+      (discussion) =>
+        new Date(discussion.time) >= new Date(start) &&
+        new Date(discussion.time) <= new Date(end),
+    );
+  }
```

Comment:
Not resolved yet!

Reply:
When you see below if else i added this code
![image](https://github.com/user-attachments/assets/b526663e-a7d7-436e-968f-bdf925331d70)

and we are passing date range 30 so it will fetch for last 30 days right?

Reply:
https://github.com/coronasafe/leaderboard/pull/463#discussion_r1682709068

Also, why perform `new Date(start` and `new Date(end)`? start and end already are date objects!

Reply:
![image](https://github.com/user-attachments/assets/47897b83-68c7-4b1a-aa9a-c68612651cab)
I hope this solve the issue

Reply:
Yup!!! This is way more readable and cleaner than before!